rocky/rocky/templates/organizations/organization_member_list.html

```diff
@@ -1,110 +1,118 @@
-            <th>{% translate 'Status' %}</th>
-            <th>{% translate 'Role' %}</th>
-            <th>{% translate 'Added on' %}</th>
-            <th>{% translate 'Has trusted permission to set OOI clearance levels' %}</th>
-            <th>{% translate 'Acknowledged permission to set OOI clearance levels' %}</th>
-            <th>{% translate 'Grant/revoke permission to set OOI clearance levels' %}</th>
-            <th>{% translate 'Edit' %}</th>
-            <th>{% translate 'Suspend' %}</th>
-            {% comment %} <th>{% translate 'Verified' %}</th> {% endcomment %}
-          </tr>
-        </thead>
-        <tbody>
-          {% for member in members %}
-            {% if not member.user.is_superuser or request.user.is_superuser %}
-              <tr>
-                <td>
-                  {% if member.user.full_name %}
-                    {{ member.user.full_name }}
-                  {% else %}
-                    {% translate 'Unset' %}
-                  {% endif %}
-                  <br>
-                </td>
-                <td>
-                  {% if member.status == "blocked" %}
-                    {% translate 'Suspended' %}
-                  {% else %}
-                    {{ member.status|title }}
-                  {% endif %}
-                  <br>
-                </td>
-                <td>
-                  {% if member.user.is_superuser %}
-                    {% translate 'Super user' %}
-                  {% endif %}
-                  {% for group in member.user.groups.all %}
-                    {{ group.name|title }}
-                    <br>
-                  {% endfor %}
-                </td>
-                <td>{{ member.user.date_joined }}</td>
-                <td>
-                  {% if member.trusted_clearance_level > 0 %}
-                    {% translate 'Yes' %} (L{{ member.trusted_clearance_level }})
-                  {% else %}
-                    {% translate 'No' %}
-                  {% endif %}
-                </td>
-                <td>
-                  {% if member.acknowledged_clearance_level > 0 %}
-                    {% translate 'Yes' %} (L{{ member.acknowledged_clearance_level }})
-                  {% else %}
-                    {% translate 'No' %}
-                  {% endif %}
-                </td>
-                <td>
-                  {% for group in member.user.groups.all %}
-                    {% for permission in group.permissions.all %}
-                      {% if 'can_set_clearance_level' in permission.codename %}
-                        {% if not member.user.is_superuser %}
-                          {% if member.trusted_clearance_level > 0 %}
-                            {% include 'partials/single_action_form.html' with btn_text="Revoke" action='withdraw_clearance' key='member_id' value=member.id btn_class="ghost" %}
+{% block content %}
+  {% include 'header.html' %}
+  <main id="main-content">
+    {% if perms.tools.view_organization  %}
```

Comment:
This looks strange to me... if someone doesn't have permissies to view this page, it seems to me that we should handle that in the view code and not have the template display an empty page.

Reply:
Using the `PermissionRequiredMixin` on the view should be the way to go.

Reply:
Thanks for the suggestion. Implemented it using the `PermissionRequiredMixin`.